framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,417 @@
+package org.checkerframework.common.value;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+/**
+ * The Range Class models an integer interval, such as all integers between 1 and 10, inclusive.
```

Comment:
The Range class documentation should indicate whether the Range class is intended to model a mathematical range or the behavior of 32-bit twos-complement integers.
I recommend 32-bit twos-complement integers, because that is more useful for use by the Value Checker.
However, even modeling 32-bit twos-complement integers requires that Range operations account for for overflow.
I have committed tests that demonstrate errors in the current implementation.
Could you fix the implementation and then re-enable the tests?  Ensuring that Range is correct will make it much easier to localize and debug problems.